packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -198,8 +198,16 @@ class AttachCommand extends FlutterCommand {
     final int devicePort = await getDevicePort();
 
     final Daemon daemon = boolArg('machine')
-      ? Daemon(stdinCommandStream, stdoutCommandResponse,
-            notifyingLogger: NotifyingLogger(), logToStdout: true)
+      ? Daemon(
+          stdinCommandStream,
+          stdoutCommandResponse,
+          notifyingLogger: NotifyingLogger(),
+          logToStdout: true,
+          // Since the application was compiled and launched by other means
```

Comment:
This comment is a little confusing, its also not quite right, since we can hot restart

Reply:
Should `attach` support `--dart-define` then? It's pretty trivial to add.

Reply:
Yes pls :)

Reply:
Done: https://github.com/flutter/flutter/pull/45317/commits/9d6a11b570ffbb0f933154826865466222a3851d